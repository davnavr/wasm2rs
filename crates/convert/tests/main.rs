//! Unit tests for code generation.

macro_rules! snapshots {
    ($(
        $name:ident($rust_capacity:literal) = $file:expr;
    )*) => {$(
        #[test]
        fn $name() {
            let wasm = wat::parse_str($file).unwrap();

            let mut rust = Vec::with_capacity($rust_capacity);
            wasm2rs_convert::Convert::new()
                .convert_from_buffer(&wasm, &mut rust)
                .unwrap();

            insta::assert_snapshot!(String::from_utf8_lossy(&rust))
        }
    )*};
}

snapshots! {
    add_5(64) = r#"(module
    (func (export "add_five") (param i32) (result i32)
        local.get 0
        i32.const 5
        i32.add))
"#;
    simple(512) = include_str!("simple.wat");
    memory(256) = include_str!("memory.wat");
    imports(256) = include_str!("imports.wat");
}
