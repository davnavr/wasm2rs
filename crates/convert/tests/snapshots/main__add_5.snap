---
source: crates/convert/tests/main.rs
expression: "String::from_utf8_lossy(&rust)"
---
// Generated by wasm2rs 0.1.0
macro_rules! wasm {
    ($vis:vis mod $module:ident use $(:: $embedder_start:ident ::)? $($embedder_more:ident)::+) => {

#[allow(non_snake_case)]
#[allow(dead_code)]
#[allow(unreachable_code)]
#[allow(unreachable_pub)]
$vis mod $module {

use $(::$embedder_start::)? $($embedder_more)::+ as embedder;
use embedder::rt::trap::TrapWith as _;

pub struct Instance {
    pub imports: embedder::Imports,
}

impl Instance {

    pub fn instantiate(store: embedder::Store) -> ::core::result::Result<embedder::Module<Self>, embedder::Trap> {
        let allocated = Self {
            imports: store.imports,
        };
        let mut module = embedder::rt::store::AllocateModule::allocate(store.instance, allocated);
        Ok(module)
    }

fn _f0(mut _l0: i32) -> i32 {
    i32::wrapping_add(_l0, 5i32)
}

pub fn add_five(&self, mut _l0: i32) -> ::core::result::Result<i32, embedder::Trap> {
    Ok(Self::_f0(_l0))
}

} // impl Instance

} // mod $module

    }
}
